### PR TITLE
fix(DB/Creature) Change creature Sacrificed Troll (14826) faction.

### PR DESCRIPTION
**Pull request by Gultask**
## Changes Proposed:
-  Changes Sacrificed Troll (14826) faction to be the same as Jin'do, so that they stop being valid targets for mind controlled players.
-  

## Issues Addressed:

- Closes unopened issue

## Tests Performed:

- Tested ingame.
- 

## How to Test the Changes:

1. `.go c 49655` with 2 or more characters. Engage Jin'do
2. wait for a Brain Wash Totem, see if the mind controlled character can see the Sacrificed Troll creatures as friendly or not (friendly = green nameplate/name)

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Nyeriah** [comment]
Hey, thanks for the PR, however faction changes need to be backed by sniffs. Do we have any sniff on this one?

**Comment by Gultask** [comment]
Thanks for pointing that out, I forgot about that.

Vmangos also has the faction set to 16, I'll open a report about it. Mind control seems to target critters a lot so the issue might be more generic than this.

**Comment by neifion-00000000** [comment]
faction itself has a lot of issues, I have a large writeup on just one of the complications

in any case I can prolly dig this data up, if I don't get back to this immediately ping me about it, I surely have them.

**Comment by neifion-00000000** [comment]
```
ServerToClient: SMSG_UPDATE_OBJECT (0x27C6) Length: 3723 ConnIdx: 1 Time: 07/01/2022 18:24:13.506 Number: 99542
NumObjUpdates: 6
MapID: 309 (309)
HasRemovedObjects: False
Data size: 3712
[0] UpdateType: CreateObject1
[0] Object Guid: Full: 0x204CB026A00E7A8000028A00003F79A6 Creature/0 R4908/S650 Map: 309 Entry: 14826 Low: 4159910
[0] Object Type: 5 (Unit)
[0] NoBirthAnim: False
[0] EnablePortals: False
[0] PlayHoverAnim: False
[0] HasMovementUpdate: True
[0] HasMovementTransport: False
[0] Stationary: False
[0] HasCombatVictim: False
[0] HasServerTime: False
[0] HasVehicleCreate: False
[0] HasAnimKitCreate: False
[0] HasRotation: False
[0] HasAreaTrigger: False
[0] HasGameObject: False
[0] HasSmoothPhasing: False
[0] ThisIsYou: False
[0] SceneObjCreate: False
[0] HasPlayerCreateData: False
[0] HasConversation: False
[0] MoverGUID: Full: 0x204CB026A00E7A8000028A00003F79A6 Creature/0 R4908/S650 Map: 309 Entry: 14826 Low: 4159910
[0] Movement Flags: 0 (None)
[0] Movement Flags 2: 0 (None)
[0] Movement Flags 3: 0 (None)
[0] MoveTime: 3155444515
[0] Position: X: -11584.153 Y: -1252.2076 Z: 77.62987
[0] Orientation: 0.959931075572967529
[0] Pitch: 0
[0] StepUpStartElevation: 0
[0] MoveIndex: 0
[0] Has Transport Data: False
[0] Has Fall Data: False
[0] HasSpline: False
[0] HeightChangeFailed: False
[0] RemoteTimeValid: False
[0] Has Inertia: False
[0] WalkSpeed: 2.5
[0] RunSpeed: 8
[0] RunBackSpeed: 4.5
[0] SwimSpeed: 4.72221994400024414
[0] SwimBackSpeed: 2.5
[0] FlightSpeed: 7
[0] FlightBackSpeed: 4.5
[0] TurnRate: 3.141592741012573242
[0] PitchRate: 3.141592741012573242
[0] MovementForceCount: 0
[0] MovementForcesModMagnitude: 1
[0] HasMovementSpline: False
[0] PauseTimesCount: 0
[0] FieldFlags: 0 (None)
[0] EntryID: 14826
[0] DynamicFlags: 0
[0] Scale: 1
[0] Health: 763
[0] MaxHealth: 763
[0] DisplayID: 14391
[0] [0] NpcFlags: 0
[0] [1] NpcFlags: 0
[0] StateSpellVisualID: 0
[0] StateAnimID: 1682
[0] StateAnimKitID: 0
[0] Charm: Full: 0x0
[0] Summon: Full: 0x0
[0] CharmedBy: Full: 0x0
[0] SummonedBy: Full: 0x0
[0] CreatedBy: Full: 0x0
[0] DemonCreator: Full: 0x0
[0] LookAtControllerTarget: Full: 0x0
[0] Target: Full: 0x0
[0] BattlePetCompanionGUID: Full: 0x0
[0] BattlePetDBID: 0
[0] (ChannelData) SpellID: 0
[0] (ChannelData) SpellXSpellVisualID: 0
[0] SummonedByHomeRealm: 0
[0] Race: 0
[0] ClassId: 1
[0] PlayerClassId: 0
[0] Sex: 2
[0] DisplayPower: 1
[0] OverrideDisplayPowerID: 0
[0] [0] Power: 0
[0] [0] MaxPower: 0
[0] [0] ModPowerRegen: 1065353216
[0] [1] Power: 0
[0] [1] MaxPower: 0
[0] [1] ModPowerRegen: 1065353216
[0] [2] Power: 0
[0] [2] MaxPower: 5
[0] [2] ModPowerRegen: 1065353216
[0] [3] Power: 1000000
[0] [3] MaxPower: 1000000
[0] [3] ModPowerRegen: 1065353216
[0] [4] Power: 0
[0] [4] MaxPower: 0
[0] [4] ModPowerRegen: 1065353216
[0] [5] Power: 0
[0] [5] MaxPower: 0
[0] [5] ModPowerRegen: 1065353216
[0] [6] Power: 0
[0] [6] MaxPower: 0
[0] [6] ModPowerRegen: 1065353216
[0] Level: 60
[0] EffectiveLevel: 0
[0] ContentTuningID: 0
[0] ScalingLevelMin: 0
[0] ScalingLevelMax: 0
[0] ScalingLevelDelta: 0
[0] ScalingFactionGroup: 0
[0] ScalingHealthItemLevelCurveID: 0
[0] ScalingDamageItemLevelCurveID: 0
[0] FactionTemplate: 16
[0] (VirtualItems) [0] ItemID: 0
[0] (VirtualItems) [0] ItemAppearanceModID: 0
[0] (VirtualItems) [0] ItemVisual: 0
[0] (VirtualItems) [1] ItemID: 0
[0] (VirtualItems) [1] ItemAppearanceModID: 0
[0] (VirtualItems) [1] ItemVisual: 0
[0] (VirtualItems) [2] ItemID: 0
[0] (VirtualItems) [2] ItemAppearanceModID: 0
[0] (VirtualItems) [2] ItemVisual: 0
[0] Flags: 32768
[0] Flags2: 2048
[0] Flags3: 0
[0] AuraState: 0
[0] [0] AttackRoundBaseTime: 2000
[0] [1] AttackRoundBaseTime: 2000
[0] BoundingRadius: 1.149999976158142089
[0] CombatReach: 1.724999904632568359
[0] DisplayScale: 1
[0] NativeDisplayID: 14391
[0] NativeXDisplayScale: 1
[0] MountDisplayID: 0
[0] StandState: 0
[0] PetTalentPoints: 0
[0] VisFlags: 0
[0] AnimTier: 0
[0] PetNumber: 0
[0] PetNameTimestamp: 0
[0] PetExperience: 0
[0] PetNextLevelExperience: 2147483647
[0] ModCastingSpeed: 1
[0] ModSpellHaste: 1
[0] ModHaste: 1
[0] ModRangedHaste: 1
[0] ModHasteRegen: 1
[0] ModTimeRate: 1
[0] CreatedBySpell: 0
[0] EmoteState: 10
[0] TrainingPointsUsed: 0
[0] TrainingPointsTotal: 0
[0] [0] ResistanceBuffModsPositive: 77
[0] [0] ResistanceBuffModsNegative: 0
[0] [1] ResistanceBuffModsPositive: 0
[0] [1] ResistanceBuffModsNegative: 0
[0] [2] ResistanceBuffModsPositive: 0
[0] [2] ResistanceBuffModsNegative: 0
[0] [3] ResistanceBuffModsPositive: 0
[0] [3] ResistanceBuffModsNegative: 0
[0] [4] ResistanceBuffModsPositive: 0
[0] [4] ResistanceBuffModsNegative: 0
[0] [5] ResistanceBuffModsPositive: 0
[0] [5] ResistanceBuffModsNegative: 0
[0] [6] ResistanceBuffModsPositive: 0
[0] [6] ResistanceBuffModsNegative: 0
[0] BaseMana: 0
[0] SheatheState: 1
[0] PvpFlags: 0
[0] PetFlags: 0
[0] ShapeshiftForm: 0
[0] HoverHeight: 1
[0] MinItemLevelCutoff: 0
[0] MinItemLevel: 0
[0] MaxItemLevel: 0
[0] WildBattlePetLevel: 0
[0] BattlePetCompanionNameTimestamp: 0
[0] InteractSpellID: 0
[0] ScaleDuration: 100
[0] LooksLikeMountID: 0
[0] LooksLikeCreatureID: 0
[0] LookAtControllerID: -1
[0] GuildGUID: Full: 0x0
[0] SkinningOwnerGUID: Full: 0x0
```

TLDR 16 seems correct

**Comment by Gultask** [comment]
@neifion-00000000 Thank You! <3

Yeah it seems that there is a problem with mind control. Not only with those skeletons but also it seems that mind controlled players target critters instead of players very often.
